scarlet/parameter.py

```diff
@@ -113,7 +113,7 @@ def is_finite(self):
 VSpace.register(Parameter, vspace_maker=VSpace.mappings[np.ndarray])
 
 
-def relative_step(X, it, factor=0.1, axis=None):
+def relative_step(X, it, factor=0.1, minimum=0, axis=None):
     """Step size set at `factor` times the mean of `X` in direction `axis`
     """
-    return factor * X.mean(axis=axis)
+    return np.maximum(minimum, factor * X.mean(axis=axis))
```

Comment:
I thought about this for a while, but it's the only way to initialize a source for zero (or very small) spectral elements without shutting it out of the optimization. Step sizes are mostly relative, so if you happen to initialize something with 0, it cannot change. The new method `set_spectra_to_match` is a linear solver, which for very faint or very blended galaxies can produce negative or zero spectra.

The spectra are set with a minimum step size proportional to the noise level in each band, which is fundamentally the uncertainty we should expect for them.

Reply:
How about initialising the source with random values of order 3 sigma of the noise or. It won't matter at this point and gradients will take off even so slightly.

Reply:
sure, we could do that, but having a minimum absolute step for a relative step size is a good idea.

Reply:
Just to be clear, this minimum step size triggers only for faint sources right? I can live with this, but otherwise, I don't see a problem with letting the algorithm take very small steps if it is not converged.

Reply:
Yes, only for faint sources.

The problem is related to #211, namely that we should have a way to determine when step sizes are too large or too small. This here one way of getting out of this problem when the initial value (and thus the step) size is initialized with or close to zero.